pkgs/tools/system/netdata/default.nix

```diff
@@ -57,18 +60,20 @@ in stdenv.mkDerivation rec {
     ''}
   '';
 
-  preConfigure = optionalString (!stdenv.isDarwin) ''
+  preConfigure = optionalString withCloud ''
+    mkdir -p externaldeps
+    cp -r "${mosquitto}/lib" externaldeps/mosquitto;
+    cp -r "${libwebsockets_3_2}/lib" externaldeps/libwebsockets;
+  '' + optionalString (!stdenv.isDarwin) ''
     substituteInPlace collectors/python.d.plugin/python_modules/third_party/lm_sensors.py \
       --replace 'ctypes.util.find_library("sensors")' '"${lm_sensors.out}/lib/libsensors${stdenv.hostPlatform.extensions.sharedLibrary}"'
   '';
 
-  configureFlags = [
-    "--localstatedir=/var"
-    "--sysconfdir=/etc"
-  ];
+  configureFlags = [ "--localstatedir=/var" "--sysconfdir=/etc" ]
+    ++ optional withCloud "--enable-cloud";
```

Comment:
Again I will ask how does this affect closure size and if it would not make sense to enable it unconditionally.

Reply:
Closure size goes from `148743560` to `149251816`, so it's basically irrelevant. 

I would be very grateful if you could tell me what the policy about adding new configuration options currently is. I generally just see the many e.g. `withCups, withDBengine, withIpmi, withNetfilter, withSsl` configuration options, many of which seem even less relevant than e.g. `withCloud`, and assume that adding new options is good practice. But your comments make me think that they might just be there for backwards compatibility and "no options" is preferred?

Reply:
I do not think there is a policy. There are few legitimate reasons for having the options overridable (when it makes the closure significantly larger, when it adds an unfree dependency to otherwise free package…) but most of the time it looks to me like a cargo-cult.

I typically ask myself: “Would someone want to disable this feature? If there were an overridable option, would someone bother overriding it? Are the benefits commensurate with the added complexity to the expression?” I would only add an option when I answered “Yes” to all three of the questions.

But in the end, the decision is up to the maintainers of each individual package, since they will bear the complexity.

Reply:
I think `withCloud` could be interesting to some users that would like to ensure that none of their system analytics are streamed to servers owned by the Netdata company. [There seems to be a way](https://github.com/netdata/netdata/issues/9043) to disable the cloud connection at runtime, but it does seem like disabling cloud support at built time is still the only option to ensure no connection is made.

Personally, I think that's useful enough to keep the `withCloud` option, which, in my opinion, also does not add a lot of complexity to the derivation.

Reply:
Looks like the cloud disabling is now fixed upstream?

Given that, I propose the package just gets built with cloud support.